cli/tools/fmt.rs

```diff
@@ -521,6 +535,138 @@ pub fn format_html(
   })
 }
 
+/// A function for formatting embedded code blocks in JavaScript and TypeScript.
+fn create_external_formatter_for_typescript(
+  unstable_options: &UnstableFmtOptions,
+) -> impl Fn(
+  MediaType,
+  String,
+  &dprint_plugin_typescript::configuration::Configuration,
+) -> Option<String> {
+  let unstable_sql = unstable_options.sql;
+  move |media_type, text, config| {
+    match media_type {
+      MediaType::Css => format_embedded_css(&text, config),
+      MediaType::Html => format_embedded_html(&text, config),
+      MediaType::Sql => {
+        if unstable_sql {
+          format_embedded_sql(&text, config)
+        } else {
+          None
+        }
+      }
+      // dprint-plugin-typescript only detects css, html, and sql for now
+      _ => unreachable!(),
```

Comment:
Maybe just return `None`? dprint-plugin-typescript could change in the future and that would break this code.

Reply:
Makes sense